src/Features/Core/Portable/ChangeSignature/DelegateInvokeMethodReferenceFinder.cs

```diff
@@ -32,29 +33,29 @@ internal class DelegateInvokeMethodReferenceFinder : AbstractReferenceFinder<IMe
         protected override bool CanFind(IMethodSymbol symbol)
             => symbol.MethodKind == MethodKind.DelegateInvoke;
 
-        protected override async Task<ImmutableArray<ISymbol>> DetermineCascadedSymbolsAsync(
+        protected override async Task<ImmutableArray<(ISymbol symbol, FindReferencesCascadeDirection cascadeDirection)>> DetermineCascadedSymbolsAsync(
             IMethodSymbol symbol,
             Solution solution,
             IImmutableSet<Project> projects,
             FindReferencesSearchOptions options,
+            FindReferencesCascadeDirection cascadeDirection,
             CancellationToken cancellationToken)
         {
-            var result = ImmutableArray.CreateBuilder<ISymbol>();
+            using var _ = ArrayBuilder<(ISymbol symbol, FindReferencesCascadeDirection cascadeDirection)>.GetInstance(out var result);
 
             var beginInvoke = symbol.ContainingType.GetMembers(WellKnownMemberNames.DelegateBeginInvokeName).FirstOrDefault();
             if (beginInvoke != null)
-            {
-                result.Add(beginInvoke);
-            }
+                result.Add((beginInvoke, cascadeDirection));
 
             // All method group references
             foreach (var project in solution.Projects)
             {
                 foreach (var document in project.Documents)
                 {
                     var changeSignatureService = document.GetLanguageService<AbstractChangeSignatureService>();
-                    result.AddRange(await changeSignatureService.DetermineCascadedSymbolsFromDelegateInvokeAsync(
-                        symbol, document, cancellationToken).ConfigureAwait(false));
+                    var cascaded = await changeSignatureService.DetermineCascadedSymbolsFromDelegateInvokeAsync(
+                        symbol, document, cancellationToken).ConfigureAwait(false);
+                    result.AddRange(cascaded.SelectAsArray(s => (s, cascadeDirection)));
```

Comment:
💡 The `SelectAsArray` isn't helping here. It would be better to either use `Select`, or an extension that allows `AddRange` with a transformation argument.